src/Http/ServerRequestFactory.php

```diff
@@ -55,35 +56,75 @@ abstract class ServerRequestFactory implements ServerRequestFactoryInterface
     public static function fromGlobals(
         ?array $server = null,
         ?array $query = null,
-        ?array $body = null,
+        ?array $post = null,
         ?array $cookies = null,
         ?array $files = null
     ): ServerRequest {
         $server = normalizeServer($server ?: $_SERVER);
         $uri = static::createUri($server);
+        $data = static::processFiles($files ?: $_FILES, $post ?: $_POST);
+        $files = $data['files'];
+        $post = $data['post'];
+
         /** @psalm-suppress NoInterfaceProperties */
         $sessionConfig = (array)Configure::read('Session') + [
             'defaults' => 'php',
             'cookiePath' => $uri->webroot,
         ];
         $session = Session::create($sessionConfig);
+
         /** @psalm-suppress NoInterfaceProperties */
         $request = new ServerRequest([
             'environment' => $server,
             'uri' => $uri,
-            'files' => $files ?: $_FILES,
+            'files' => $files,
             'cookies' => $cookies ?: $_COOKIE,
             'query' => $query ?: $_GET,
-            'post' => $body ?: $_POST,
+            'post' => $post,
             'webroot' => $uri->webroot,
             'base' => $uri->base,
             'session' => $session,
-            'mergeFilesAsObjects' => Configure::read('App.uploadedFilesAsObjects', true),
         ]);
 
         return $request;
     }
 
+    /**
+     * Process uploaded files and move things onto the post data.
+     *
+     * @param array $files Files array for normalization and merging in post body.
+     * @param array $post Post body to merge files onto.
+     * @return array
+     * @psalm-return array{files: array, post: array}
+     */
+    protected static function processFiles($files, $post)
+    {
+        $files = normalizeUploadedFiles($files ?: $_FILES);
```

Comment:
:clap: Nice that we don't have to maintain this anymore. I like that you kept a few integration tests to ensure laminas keeps their end of the contract up.